File: Modify-the-Matrix.py
class Solution(object):
    def modifiedMatrix(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[List[int]]
        """
        m, n = len(matrix), len(matrix[0])

        ans = [[0]*n for _ in range(m)]
        col = [-1]*n

        for i in range(n):
            for j in range(m):
                ans[j][i] = matrix[j][i]
                col[i] = max(col[i], matrix[j][i])

        for i in range(n):
            for j in range(m):
                if ans[j][i] == -1:
                    ans[j][i] = col[i]

        return ans

File: Modify-the-Matrix.cpp
class Solution {
public:
    vector<vector<int>> modifiedMatrix(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();

        vector<vector<int>> ans(m, vector<int>(n, 0));
        vector<int> col(n, -1);

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                ans[j][i] = matrix[j][i];
                col[i] = max(col[i], matrix[j][i]);
            }
        }

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (ans[j][i] == -1) {
                    ans[j][i] = col[i];
                }
            }
        }

        return ans;
    }
};
